### PR TITLE
secborgs behind code blue

### PR DESCRIPTION
**Pull request by SirSkeleto**
[balance]

Oh boy.

:cl:
 * rscadd: The security cyborg module is now locked behind code blue and above.

### CODE REVIEW

**Comment by boy2mantwicethefam** [comment]
Based

**Comment by iFlashYou** [comment]
so brave

**Comment by aceedex** [comment]
Agh no borg please my jecties Im almost to the shuttle
Borg uncuff me and go away nooooo

**Comment by Kammerjunk** [comment]
isn't it like only a year ago we last did this

**Comment by Hoshi-chan** [comment]
This _might_ force security to actually adhere to the code system. Not because they care about having secborgs, but because validhunting silicons who need to go sec will nag them to raise the code so that they can.

On the other hand, this is sort of like having the armory forcibly locked down by blast doors that only open on code blue.

Gonna have to lean towards "bad idea," though, because borgs are already cucked so who cares.

**Comment by Ephx1** [comment]
ahahahahahahaHAHAHAHAHAHAHA
_AHAHAHAHAHAHAHAHAHAHAHAHAHA_

**Comment by brndd** [comment]
This would be alright if you didn't have to go through a 5+ minute ordeal of tracking down the roboticist and getting him to install the fucking reset board every time you change modules. And that's assuming there even is a roboticist.

Borgs should be able to change modules at will in rechargers.

**Comment by SonixApache** [comment]
Didn't work last time, won't work this time.
No motivation given either. Why do you think we should do this? Why do you think now will be different from last time we tried it?

**Comment by Hoshi-chan** [comment]
@brndd Please code that. It sounds great.

**Comment by ghost** [comment]
All this will do is encourage code blue at start of round.

**Comment by iFlashYou** [comment]
nah, it will not

**Comment by FudgePucker** [comment]
What about malf?

**Comment by TurboGay** [comment]
This is also a nerf to traitor/malfunctioning AI. Still a bad idea even if you somehow manage to address that problem and all the other issues this would cause. And as others have asked, what's the justification behind this PR?

**Comment by ghost** [comment]
I never play borgs and primarily play science, let me tell you that reset board mechanic is terrible and borgs should be able to pick their module in a recharging station before this change is even considered.

To understand why reset is terrible: currently a borg changing their module will wipe all their upgrades, meaning they need to be printed again. It will also consume the reset board.

**Comment by Coxswain-Navigator** [comment]
Finally a pr everyone can agree on.

**Comment by DrSnips** [comment]
Like sonix said we already tried this with #14488 and then reverted it soon after with #14516 and #14762 because people didn't like it. It's already proven to be a disliked change.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
If people can point to a 64/36 poll and say "no that doesn't count because it's 2 years old", surely we can also rule out a 2 year old experiment then.

**Comment by aceedex** [comment]
Dont think anyone should do that considering the OP doesnt even have any reasoning for this change. Just another of the myriad "I dislike thing so remove it" PRs

**Comment by Lunstra** [comment]
Gay though sec borgs are this doesn't do shit with the real problems with them, namely; Asimov sec borgs aka "Borg law 2 uncuff me then go away", the mindset of people who actually choose to play as a validhunterbot meaning they'd just do something like play mediborg with an intercom on the sec frequency while they watch the CMC, the first of which is a Feature™ and the second is unfixable without big honking policy change that'd shift us closer to high RP which nobody really wants. Oh AND this would also hard dicks malf which in my experience usually gets shit on anyway.

**Comment by ShiftyRail** [comment]
I made a previous PR about this a little over a year ago (already ? Time goes by so fast).
#17332

Here is the reasoning I had at the time.

The secborgs contribution to the vast majority of rounds is a negative one. Even now with malf being much rarer and traitor AI being gone, the scars are still here.

Point 1 : **Security cyborgs are a very powerful tool that should be locked behind special circumstances.**
Borgs in general are something you can hardly trust as a mere civilian, because of how often they get subverted. Secborgs are even worst in that fashion because it's not easy to reliably stun them (you can get a blinder but that requires effort and preparation), and they have a virtually infinite taser, cuffs, and a harmbaton.
While you shouldn't feel safe 24/7 in spess, of course, this is the kind of frustration that's not fun.
**If** this PR is merged, this would go a long way to solve that. If there's a secborg out, I know it's code blue and that security is (probably) aware of it.

Point 2 : **Security cyborgs in peaceful times are, simply speaking, annoying.**

It's not that they're overpowered, impossible to deal with, or something of the sort.
I know that you can order a borg to change its module to non-sec if you've got 3 officers and no doctors, I know you can order him to "keep away from brig unless there's human harm".

The problem is that the last time someone did that, they got ahelped and had a note, so clearly this isn't something most people will want to do.
Even though that note got removed, it can still cause people to have second thoughts about ordering a borg to change modules.

Secborgs are annoying because they're designed to apprehend and restrain criminals, but because of the way laws work, they're terribly ineffective at it and just end up either selectively listening to orders or just getting in your way, all the time, until you change the lawset, which I believe shouldn't be a common occurrence.

>It's a problem with Asimov!!!!!!!

No, it's not ; NT Default would heavily emphasise problem 1 (and is by far the worst lawset anyways). NT Default gives the Captain/HoS/whatever absolute authority roundstart over the silicons, which is not healthy for round progression at all. It's fashy future incarnate.
Robocop is out of the question.
Any other lawset (Paladin, Corpororate...) is just a joke because it will give them free griff licence.

Point 3 : **Secborgs and round progression**

While dynamic mode and his abundance of antags made this point a little moot, I don't believe a unstunable machine with an infinite taser, immune to slips and with all-access, is healthy to have roundstart, especially when it's so easy to subvert.

TL;DR : I don't believe sec borgs make a positive contribution to rounds.

**Comment by boy2mantwicethefam** [comment]
>Secborgs finally locked behind code blue
>People come out of the woodwork to be against it for whatever reason

I should ungunk malf

**Comment by Kammerjunk** [comment]
@aceedex our system for code changes discourages PR authors from arguing for their change because a PR is by default "good" and you can therefore only hurt yourself by trying to justify your PR, even if it's Barry-tier
Everything in this post will be with the assumption that (You) haven't read the last thread because like I said just above, why would you. don't expect any of this to be read either because again, there's absolutely zero incentive to
Because the PR just says "oh boy," a lot of this is going to assume a security standpoint - the last PR was from a security standpoint (as said in one of the comments, not the OP) and it's certainly rare that I've had problems with them when doing normal duties in medbay, cargo, engineering, science, botany, or as assistant.

- Unless you plan to remove the headsets/intercoms of all criminals or suspected criminals (and I assure you this won't fix the problem), this won't stop "borg law 2 uncuff me/release me/stun that officer" just as it won't stop "HELP BEING EXECUTED IN SEC." Unless you're implying that borgs should covertly or conveniently "miss" lines like that over comms, they're still required by Asimov to obey human orders and they're still required by Asimov to a) not harm humans, b) stop human harm, and c) prevent probable (read: almost guaranteed) human harm - they've just lost their best method of doing so.
- Security, along with medical, are the modules that most appropriately fit the Asimov laws (and we do still assume the Asimov laws). The way the borg and silicon systems work, it's not exactly the case that "janitor borgs are janitor robots, miner borgs should are miner robots, medborgs are doctor robots, and secborgs should are security officer robots." Those are all things they *do* (or rather things their players want to do), but their main functions are still, in order of importance, 1) stopping, preventing, and not causing human harm; 2) obeying human orders, and 3) protecting their own existence. As long as they observe their laws and server rules, they're free to play with their toolset.
Security and medical modules do this better than all others, through stopping either *self-contained* human harm (e.g. toxins from a damaged liver) or *external* harm (e.g. being beaten up by another human).
Importantly, they're able to non-harmfully prevent *serial* human harmers from being able to harm again. That bit is important because once a human has shown that they'll keep harming humans if capable, the First Law demands that they're (non-harmfully) prevented from being *able* to harm, not just prevented from harming.
- As an extension of the above, Asimov has nothing to do with alert codes. As the station-wide (or world-wide?) activation message says, code blue is justified by "reliable information about possible hostile activity" - not dangerous, but hostile. It's not much an Asimov borg's concern if someone is there to steal a hand tele.

- Ops in general. I said as much in the last PR thread for locking secborgs behind code blue, but you've never gotten very much time to do things that aren't stopping the ops. Upping to code blue for borgs is pretty low on that list, which sounds frustrating for the borgs when the security module is the _single best way to enforce Asimov._ 
- also fucks malf for obvious reasons, which is relevant regardless of how rare it is


lastly, in order to not become my worst enemy Nuke:
### TL;DR if the exact same problems for security happen without secborgs, perhaps secborgs aren't the problem. in ops, this creates harmdragging gas except for all borgs instead of just medborgs
in memoriam Aurx:
image

**Comment by ShiftyRail** [comment]
>dude just change the AI's laws

What if we didn't have to do that every round

**Comment by Kammerjunk** [comment]
"if the exact same problems for security happen without secborgs, perhaps secborgs aren't the problem"

**Comment by jknpj** [comment]
For everyone ranting about module resets being a pain in the ass and flawed: Robots/Cyborgs aren't omni-machines like mommis.

The fact that module resets are super painful to get and their upgrades are lost IS intended, they were also NOT designed around being able to change modules on the go. It's the robo's duty to perform silicon maintenance and unless you make recharging stations require a physical reset board to be inserted inside them(like 'automatic' cell changes are made) i don't think it will be a good change.

Imagine if we replaced the HOP with a machine that you could put your ID inside and change your job without having to deal with the pain of asking for it. That would be great in 'emergencies', right? This is basically what you're asking for, but for cyborgs.

**Comment by brndd** [comment]
Naturally it's *intended* that the module resets work the way they do. It's an entirely different question whether that intention leads to good gameplay.

I agree it could be silly if borgs constantly change their module, but if this PR were to go through security borgs will become borderline useless. By the time they can get the reset board in, more often than not the active threat is already over. It's different from red alert because that's usually raised for bleb/xenos/cult etc. and those threats last long enough that getting the reset board to go into combat is worthwhile. Blue alert just isn't used like that in practice, even though if you go by in-character logic then blue alert would be raised as soon as any sort of antagonistic activity can reasonably be suspected, and it would stay on until the suspects are contained.

Maybe borgs should get *one* free module change in a recharger whenever the alert level changes, and this doesn't stack, and the opportunity goes away if the alert is canceled before the borg uses it.

**Comment by ghost** [comment]
@jknpj
>The fact that module resets are super painful to get and their upgrades are lost IS intended

Appeal to tradition, every past balance change has altered established intent in favor of new intent. If a change goes through to make borgs able to select their modules at a recharger, it would become the new intended way. This line of argumentation has no merit.

>Imagine if we replaced the HOP with a machine that you could put your ID inside and change your job without having to deal with the pain of asking for it. That would be great in 'emergencies', right? This is basically what you're asking for, but for cyborgs.

False equivalence. "Humans" can pickup and use any item and therefor are not limited by access restrictions despite lacking the proper ID, except for some console operations which are negligible in context of gameplay being argued here.

**Comment by tayachiya** [comment]
Probably was asked before, but I have to ask yet again - how do you non-harmfully stop anyone wearing sunglasses from murdering now? Flashing them won't work, can't reliably use smoke because they can either simply move away, or you'll get a lot of other people as well and end up being bwoinked. Maybe a chloral syringe, but that is one time thing if you harm-stab as medborg, which would require a module reset to fix. 
And everyone always says how secborgs don't follow space law, yet code blue is very much a space law related event. Those two things should not be connected.

**Comment by jknpj** [comment]
>Appeal to tradition

It's not. I said they weren't designed with being able to switch modules on their own, this is beyond a "old good, change... bad." mentality. If we want to change that, we also would need to change their very design to match the new choices.

IMAGINE all cyborgs rushing to hotswap to security the instant someone screams "HELP MAINT" and how this will affect both round progression and how people deal with silicons. As they are RIGHT NOW it would be the nuts.

>"Humans" can pickup and use any item and therefor are not limited by access restrictions

:thinking:

**Comment by ghost** [comment]
How are they going to swap modules in time when they first have to get to a recharger, select the module and wait for the swap duration? Even if such a delay would still seem too short, a module swap could drain battery requiring further "timeout". There's many ways to balance such a feature.

I'm not going to humor emoji responses, put effort into your argument or don't bother.

**Comment by SonixApache** [comment]
something something fallacy fallacy

**Comment by jknpj** [comment]
>I'm not going to humor emoji responses

I was simply unable to make sure if you were being serious or not. It's just too pretentious to say that  access restrictions are "negligible in context of gameplay". 

The sole reason why silicons are so powerful is EXACTLY their all access. Being able to control machinery remotely means nothing is you have no access to it, after all.

**Comment by aceedex** [comment]
Sorry, if the issues is that secborgs follow asimov too much then this PR aint the way to fix it. Id downright say that's a completely IC issue.

If a secborg, who by default is primarily concerned with asimov, keeps releasing prisoners tell it to stay out of the brig and go prevent harm somewhere else

**Comment by ghost** [comment]
@jknpj
Modules changes have no relation whatsoever to access restrictions, since they do not change access and only allow usage of different items. The false equivalence comes from you comparing item usage to access, using it as an argument for why module changes would be a mistake. They are completely different things.

Off the top of my head, these are the only ID locked interactions a "human" player cannot bypass without an ID:
* Call emergency shuttle
* Change ID access
* Sync RND console
* Operate Exosuit fabricator
* Initiate maintenance mode for mechs
* Send/Receive shuttles
* Change alert status
* Announcements

None of the above are related to security module or offer any notable advantage in situations where sec borgs are perceived as a problem. Their only advantage when it comes to access is faster response times due to not having to hack doors, which is again not related to module changes.

**Comment by ShiftyRail** [comment]
@kammer Security borgs are a different can of worms than engineering or the regular old borgos because they're more frustrating to handle, and tend to wander around brig more.

Also sorry random person I @ 'd by mistake

**Comment by Kammerjunk** [comment]
any differences between secborg silicons and non-secborg silicons regarding letting criminals out of the brigs/cucking sec from executing are not so great that this will fix the issue.
considering this has been a thing on this server for as long as this server has existed, it is not so critical an issue that a halfway solution like this is necessary - especially because a halfway solution will make it less likely that someone becomes motivated to push through a full solution

in short, this PR doesn't solve the problem it (presumably) sets out to fix and it makes similar problems less likely to be solved because it's "good enough"

**Comment by SonixApache** [comment]
For the record, borgs (of any kind) aren't an issue for security because they get all the necessary tools to stun one, and their IDs unlock them.

I'm still waiting for the motivation for this PR.

**Comment by ShiftyRail** [comment]
>and their IDs unlock them.

Hold on a second, since when

**Comment by jknpj** [comment]
That PR didn't pass

**Comment by PJB3005** [comment]
Clearly you should've just listened to me when I pushed for baby blue peacekeeper borgs.

**Comment by Kammerjunk** [comment]
god shut up PJB I hate not disagreeing with collabs

**Comment by SonixApache** [comment]
I don't know since when, but it's been a thing for months. It's how I keep disassembling roundstart secborgs that join teams of 3+ officers and refuse to get their module changed.

**Comment by ShiftyRail** [comment]
>in short, this PR doesn't solve the problem it (presumably) sets out to fix and it makes similar problems less likely to be solved because it's "good enough"

1) It will solve the most common offender - there's a reason why sec borgs are more hated than other modules. And that's because their anti-fun potential is much higher.

2) Doesn't convince me at all. This PR works for what it tries to do.

**Comment by SonixApache** [comment]
>This PR works for what it tries to do.

I still don't know what it tries to do.

**Comment by Kammerjunk** [comment]
1. sounds like a shitty perception and also I'm going to doubt you when you say secborg and not AI is the "most common offender." this PR does not solve the issue it sets out to solve
2. relies on 1
3. read the bit about criticality again

**Comment by agentofzog2** [comment]
All the anger over secborgs is because they're red, if they were painted more calming color, like blue, everything would be fine,

**Comment by Hoshi-chan** [comment]
> 2. Doesn't convince me at all. This PR works for what it tries to do.

And what, exactly, is it trying to do? There is still no stated reason for this PR.

**Comment by Blithering** [comment]
Security module isn't the problem. 
Asimov isn't the problem.
Security isn't the problem.
People calling law 2 every time they get arrested isn't the problem.
Lack of muzzles in cell lockers is the problem.

**Comment by ShiftyRail** [comment]
>And what, exactly, is it trying to do?

Make sec borgs less present in the game, and present only when their presence is warranted for sure. Which I feel is needed as it is my belief that sec borgos are a net negative contribution to the game.

**Comment by brndd** [comment]
Isn't the problem that secborg players are shitters who hound security dragging greyshirts away and releasing prisoners in a thinly veiled attempt to make security set them to robocop so they can validhunt?

Just start bwonking people for that shit. Asimov secborgs should be standing watch in public areas and breaking up fights and do very little else. Also tasering isn't against law 1 because law 1 says they can't injure people (*harm* is only for the inaction clause!) and tasers do no damage. I'll stop silicon lawyering now.

**Comment by Kammerjunk** [comment]
@ShiftyRail not to sound like M*lyneux, but nothing in that post argues for this PR

File: __DEFINES/silicon.dm [hunk 1]
```diff
@@ -50,8 +53,14 @@ var/global/list/all_robot_modules = default_nanotrasen_robot_modules + emergency
 
 /proc/getAvailableRobotModules()
 	var/list/pickable_modules = default_nanotrasen_robot_modules.Copy()
-	if(security_level == SEC_LEVEL_RED)
-		pickable_modules += emergency_nanotrasen_robot_modules
+	switch(security_level)
+		if(SEC_LEVEL_BLUE)
```
Comment (jknpj) [inline __DEFINES/silicon.dm:57]: if(security_level >= SEC_LEVEL_BLUE)
	pickable_modules += alert_nanotrasen_robot_modules
if(security_level >= SEC_LEVEL_RED)
	pickable_modules += emergency_nanotrasen_robot_modules

**Review by jknpj — CHANGES_REQUESTED** [review]

**Comment by SirSkeleto** [comment]
going to close this until i'm done with my final project so I can get around to coding a way that borgs can change their own module in a recharger.

**Comment by jknpj** [comment]
@SirSkeleto i'm literally working on this
